### PR TITLE
[fix]: [PL-27908]: Custom sm host machine changes

### PR DESCRIPTION
**Pull request by shreyas-harness**
## Describe your changes
Adding capablity to run on host machine
## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/37935)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 No issues Everything looks good
 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/37935)

File: 930-delegate-tasks/src/main/java/software/wings/service/impl/security/customsecretsmanager/CustomSecretsManagerValidationUtils.java [hunk 1]
```diff
@@ -138,14 +138,20 @@ public static ShellScriptTaskParametersNG buildShellScriptTaskParametersNG(Strin
     ScriptType scriptType = ScriptType.BASH;
     ShellScriptTaskParametersNGBuilder taskParametersNGBuilder = ShellScriptTaskParametersNG.builder();
     Map<String, String> envVars = new HashMap<>();
+    boolean executeOnDelegate = Boolean.TRUE.equals(customSecretNGManagerConfig.isOnDelegate());
+    String executionId = "custom-sm"
```
Comment (meenakshi-harness) [inline 930-delegate-tasks/src/main/java/software/wings/service/impl/security/customsecretsmanager/CustomSecretsManagerValidationUtils.java:142]: Why we need this?

**Comment by shreyas-harness** [comment]
trigger messagecheck

**Review by shreyas-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37935)* status: 13 file(s) to review. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`930-delegate-tasks/src/main/java/software/wings/service/impl/security/customsecretsmanager/CustomSecretsManagerValidationUtils.java` line 142 at r2](https://reviewable.io/reviews/harness/harness-core/37935#-NCuUF8wRF5l6ihFYDFu-r2-142:-NCyrPp74SnuRDpLiiBh:b-uq76ag) ([raw file](https://github.com/harness/harness-core/blob/2ed6145eb1eb084bc562f81ac31122b5c0aabc17/930-delegate-tasks/src/main/java/software/wings/service/impl/security/customsecretsmanager/CustomSecretsManagerValidationUtils.java#L142)):*
<details><summary><i>Previously, meenakshi-harness (Meenakshi Raikwar) wrote…</i></summary><blockquote>

Why we need this?
</blockquote></details>

Execution id needs to be a not null value. 
Setting it as custom-sm-{timestamp} for easy navigation in logs.

File: 930-delegate-tasks/src/main/java/software/wings/service/impl/security/customsecretsmanager/CustomSecretsManagerValidationUtils.java (hunk 1 above)
Comment (shreyas-harness) [reply 930-delegate-tasks/src/main/java/software/wings/service/impl/security/customsecretsmanager/CustomSecretsManagerValidationUtils.java:142]: Also, the execution id is used for caching session which we haven't enabled. So any not-null value would solve the purpose

Comment (meenakshi-harness) [reply 930-delegate-tasks/src/main/java/software/wings/service/impl/security/customsecretsmanager/CustomSecretsManagerValidationUtils.java:142]: Is it required in Params DTO or in Handler? If in handler than lets modify the code to not use it?

**Review by meenakshi-harness — CHANGES_REQUESTED** [review]

**Review by shreyas-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37935)* status: 14 file(s) to review. 2 change requests. 3. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`890-sm-core/src/main/java/io/harness/ng/core/encryptors/NGManagerCustomEncryptor.java` line 126 at r5](https://reviewable.io/reviews/harness/harness-core/37935#-ND1yeEE4QLu6JDYcPfL:-ND3msy2EFvBYcN2gVYT:b-otmhrv) ([raw file](https://github.com/harness/harness-core/blob/143863a5515a02a7781c23375af4d2fe96dd5e2d/890-sm-core/src/main/java/io/harness/ng/core/encryptors/NGManagerCustomEncryptor.java#L126)):*
<details><summary><i>Previously, nishantsaini20 (Nishant Saini) wrote…</i></summary><blockquote>

Add UT covering this method
</blockquote></details>

Done. Added test case
___
*[`890-sm-core/src/main/java/io/harness/ng/core/encryptors/NGManagerCustomEncryptor.java` line 137 at r5](https://reviewable.io/reviews/harness/harness-core/37935#-ND1xvDD-jPgRwPDh87j:-ND3mvMxBd8isKZB-7Bg:b-9ulnoc) ([raw file](https://github.com/harness/harness-core/blob/143863a5515a02a7781c23375af4d2fe96dd5e2d/890-sm-core/src/main/java/io/harness/ng/core/encryptors/NGManagerCustomEncryptor.java#L137)):*
<details><summary><i>Previously, nishantsaini20 (Nishant Saini) wrote…</i></summary><blockquote>

Check that errorMSg as a parameter doesn't suppress original message.
</blockquote></details>

It does not suppress the original message. Only if we dont get any message but still null result, we use it as default message. 
___
*[`930-delegate-tasks/src/main/java/software/wings/service/impl/security/customsecretsmanager/CustomSecretsManagerValidationUtils.java` line 142 at r2](https://reviewable.io/reviews/harness/harness-core/37935#-NCuUF8wRF5l6ihFYDFu-r2-142:-ND3n3d16ybR-eU4P3Fb:bfbtml9) ([raw file](https://github.com/harness/harness-core/blob/2ed6145eb1eb084bc562f81ac31122b5c0aabc17/930-delegate-tasks/src/main/java/software/wings/service/impl/security/customsecretsmanager/CustomSecretsManagerValidationUtils.java#L142)):*
<details><summary><i>Previously, meenakshi-harness (Meenakshi Raikwar) wrote…</i></summary><blockquote>

Is it required in Params DTO or in Handler? If in handler than lets modify the code to not use it?
</blockquote></details>

Made it random uuid since any random value will work for us.

**Review by nishantsaini20 — CHANGES_REQUESTED** [review]
> Reviewed 2 of 3 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37935)* status: 14 file(s) to review. 2 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`890-sm-core/src/main/java/io/harness/ng/core/encryptors/NGManagerCustomEncryptor.java` line 128 at r6](https://reviewable.io/reviews/harness/harness-core/37935#-ND6OQ7r7XFrXLxCq_Zb:-ND6OQ7r7XFrXLxCq_Zc:b7ecfqg) ([raw file](https://github.com/harness/harness-core/blob/de6257cdde44ea46d397f7be161cc3c599530dda/890-sm-core/src/main/java/io/harness/ng/core/encryptors/NGManagerCustomEncryptor.java#L128)):*
> ```Java
> 
>   @VisibleForTesting
>   public void getSSHSupportedConfig(EncryptionConfig encryptionConfig) {
> ```

This method doesn't return anything. It should be renamed to something like fillSSHSupportedConfig or addSSHSupportedConfig
___
*[`890-sm-core/src/test/java/io/harness/encryptors/NGManagerCustomEncryptorTest.java` line 76 at r6](https://reviewable.io/reviews/harness/harness-core/37935#-ND6Nqhj1EGZFM3g2V_v:-ND6Nqhk3uU6FFWKcZjD:b-iw90by) ([raw file](https://github.com/harness/harness-core/blob/de6257cdde44ea46d397f7be161cc3c599530dda/890-sm-core/src/test/java/io/harness/encryptors/NGManagerCustomEncryptorTest.java#L76)):*
> ```Java
>     assertThat(customSecretNGManagerConfig.getSshKeyEncryptionDetails()).isNotNull();
>     assertThat(customSecretNGManagerConfig.getSshKeyEncryptionDetails()).isEqualTo(encryptedDataDetails);
>   }
> ```

UT missing for case when secretResponseWrapper is null.

**Review by shreyas-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37935)* status: 14 file(s) to review. 2 change requests. 3. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`890-sm-core/src/main/java/io/harness/ng/core/encryptors/NGManagerCustomEncryptor.java` line 128 at r6](https://reviewable.io/reviews/harness/harness-core/37935#-ND6OQ7r7XFrXLxCq_Zb:-ND6ZAdH89LbaHPmCC-i:bctnihh) ([raw file](https://github.com/harness/harness-core/blob/de6257cdde44ea46d397f7be161cc3c599530dda/890-sm-core/src/main/java/io/harness/ng/core/encryptors/NGManagerCustomEncryptor.java#L128)):*
<details><summary><i>Previously, nishantsaini20 (Nishant Saini) wrote…</i></summary><blockquote>

This method doesn't return anything. It should be renamed to something like fillSSHSupportedConfig or addSSHSupportedConfig
</blockquote></details>

Updated to method name to addSSHSupportedConfig
___
*[`890-sm-core/src/test/java/io/harness/encryptors/NGManagerCustomEncryptorTest.java` line 76 at r6](https://reviewable.io/reviews/harness/harness-core/37935#-ND6Nqhj1EGZFM3g2V_v:-ND6ZExO1_RBWTNwSNSO:b-b8rdhf) ([raw file](https://github.com/harness/harness-core/blob/de6257cdde44ea46d397f7be161cc3c599530dda/890-sm-core/src/test/java/io/harness/encryptors/NGManagerCustomEncryptorTest.java#L76)):*
<details><summary><i>Previously, nishantsaini20 (Nishant Saini) wrote…</i></summary><blockquote>

UT missing for case when secretResponseWrapper is null.
</blockquote></details>

Added test case to check correct exception is thrown when secret response data is null

**Review by nishantsaini20 — DISMISSED** [review]
> Reviewed 2 of 2 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37935)* status: 14 file(s) to review. 1 change requests. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by meenakshi-harness — DISMISSED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37935)* status: 14 file(s) to review.

**Review by nishantsaini20 — APPROVED** [review]
> Reviewed 1 of 1 files at r8, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37935)* status: 14 file(s) to review. 1 of 2 approvals obtained.

**Review by meenakshi-harness — APPROVED** [review]
> Reviewed 4 of 9 files at r1, 5 of 6 files at r2, 1 of 1 files at r5, 1 of 3 files at r6, 2 of 2 files at r7, 1 of 1 files at r8, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37935)* status: :shipit: complete! ✓

**Comment by piyush-harness** [comment]
Approved for hashcheck

**Review by piyush-harness — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37935#-:-ND8grq2A386tIBZDtOg:bnfp4nl)* status: :shipit: complete! ✓

**Pull request merged**